src/Perlang.Parser/Scanner.cs

```diff
@@ -339,32 +351,37 @@ private void Number()
             }
             else
             {
+                string numberCharacters = source[(start + startOffset)..current];
+
                 // Any potential preceding '-' character has already been taken care of at this stage => we can treat
                 // the number as an unsigned value. However, we still try to coerce it to the smallest signed or
                 // unsigned integer type in which it will fit (but never smaller than 32-bit). This coincidentally
                 // follows the same semantics as how C# does it, for simplicity.
-                BigInteger value;
 
-                if (numberBase == Base.HEXADECIMAL)
-                {
-                    string numberCharacters = source[(start + startOffset)..current];
-
-                    // Quoting from
-                    //https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.parse?view=net-5.0#System_Numerics_BigInteger_Parse_System_ReadOnlySpan_System_Char__System_Globalization_NumberStyles_System_IFormatProvider_
-                    //
-                    // If value is a hexadecimal string, the Parse(String, NumberStyles) method interprets value as a
-                    // negative number stored by using two's complement representation if its first two hexadecimal
-                    // digits are greater than or equal to 0x80. In other words, the method interprets the highest-order
-                    // bit of the first byte in value as the sign bit. To make sure that a hexadecimal string is
-                    // correctly interpreted as a positive number, the first digit in value must have a value of zero.
-                    //
-                    // We presume that all hexadecimals should be treated as positive numbers for now.
-                    value = BigInteger.Parse('0' + numberCharacters, numberStyles);
-                }
-                else
+                BigInteger value = numberBase switch
                 {
-                    value = BigInteger.Parse(source[(start + startOffset)..current], numberStyles);
-                }
+                    Base.DECIMAL =>
+                        BigInteger.Parse(source[(start + startOffset)..current], numberStyles),
+
+                    Base.BINARY =>
+                        Convert.ToUInt64(numberCharacters, 2),
+
+                    Base.HEXADECIMAL =>
+                        // Quoting from
```

Comment:
Filed an upstream issue about this (seemingly) false positive error: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3392

I'll suppress this using the approach provided in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1515.md for now.